helm/cloud-provider-azure/templates/cloud-provider-azure.yaml

```diff
@@ -0,0 +1,487 @@
+            {{- end }}
+            {{- if hasKey .Values.cloudNodeManager "master" }}
+            - "--master={{ .Values.cloudNodeManager.master }}"
+            {{- end }}
+            {{- if hasKey .Values.cloudNodeManager "minResyncPeriod" }}
+            - "--min-resync-period={{ .Values.cloudNodeManager.minResyncPeriod }}"
+            {{- end }}
+            {{- if hasKey .Values.cloudNodeManager "nodeStatusUpdateFrequency" }}
+            - "--node-status-update-frequency={{ .Values.cloudNodeManager.nodeStatusUpdateFrequency }}"
+            {{- end }}
+            {{- if hasKey .Values.cloudNodeManager "useInstanceMetadata" }}
+            - "--use-instance-metadata={{ .Values.cloudNodeManager.useInstanceMetadata }}"
+            {{- end }}
+            {{- if hasKey .Values.cloudNodeManager "waitRoutes" }}
+            - "--wait-routes={{ .Values.cloudNodeManager.waitRoutes }}"
+            {{- end }}
+          env:
+            - name: NODE_NAME
+              valueFrom:
+                fieldRef:
+                  fieldPath: spec.nodeName
+          resources:
+            requests:
+              cpu: {{ .Values.cloudNodeManager.containerResourceManagement.requestsCPU }}
+              memory: {{ .Values.cloudNodeManager.containerResourceManagement.requestsMem }}
+            limits:
+              cpu: {{ .Values.cloudNodeManager.containerResourceManagement.limitsCPU }}
+              memory: {{ .Values.cloudNodeManager.containerResourceManagement.limitsMem }}
+---
+apiVersion: apps/v1
+kind: DaemonSet
+metadata:
+  name: cloud-node-manager-windows
+  namespace: kube-system
+  labels:
+    component: cloud-node-manager
+    kubernetes.io/cluster-service: "true"
+spec:
+  selector:
+    matchLabels:
+      k8s-app: cloud-node-manager-windows
+  template:
+    metadata:
+      labels:
+        k8s-app: cloud-node-manager-windows
+      annotations:
+        cluster-autoscaler.kubernetes.io/daemonset-pod: "true"
+    spec:
+      priorityClassName: system-node-critical
+      serviceAccountName: cloud-node-manager
+      nodeSelector:
+        kubernetes.io/os: windows
+      tolerations:
+      - key: CriticalAddonsOnly
+        operator: Exists
+      - key: node-role.kubernetes.io/master
+        operator: Equal
+        value: "true"
+        effect: NoSchedule
+      - operator: "Exists"
+        effect: NoExecute
+      - operator: "Exists"
+        effect: NoSchedule
+      containers:
+      - name: cloud-node-manager
+        image: {{ template "image.cloudNodeManager" . | required "you must use a supported version of Kubernetes or provide cloudNodeManager.imageRepository, cloudNodeManager.imageName, and cloudNodeManager.imageTag values" }}
+        imagePullPolicy: {{ .Values.cloudNodeManager.imagePullPolicy }}
+        command: ["/cloud-node-manager.exe"]
+        args:
+          - --node-name=$(NODE_NAME)
+          {{- if hasKey .Values.cloudNodeManager "cloudConfig" }}
```

Comment:
@feiskyer @lzhecheng Windows cloud-node-manager daemonset added. Lemme know if any of the below configurable options (`--cloud-config`, `--kube-api-burst`, `--kube-api-content-type`, etc) are not relevant for the windows `cloud-node-manager.exe` runtime.